core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -193,7 +194,10 @@ private InputStream openAtPosition(long position) throws IOException {
         }
         UnderFileSystem ufs = UnderFileSystem.get(mUri);
         // TODO(calvin): Consider making openAtPosition part of the UFS API
-        if (ufs instanceof S3UnderFileSystem) { // Optimization for S3 UFS
+        if (ufs instanceof S3AUnderFileSystem) { // Optimization for S3A UFS
+          mStream =
+              new CountingInputStream(((S3AUnderFileSystem) ufs).openAtPosition(mUri, position));
```

Comment:
in the other `else` blocks, it does `mInitPos = position`. This block doesn't need that line?